### ISSUE TITLE
OnWorldLoaded impulse

### ISSUE THREAD

**Issue by 3x1t-5tyl3** [Neos-Metaverse/NeosPublic#3041]
### Is your feature request related to a problem? Please describe.

Currently we have to test against T, which works, but I felt like a proper implementation is a good idea.

### Relevant issues

-

### Describe the solution you'd like

An event impulse "OnWorldLoaded" which would send an impulse on world load..

### Describe alternatives you've considered

Test with a "OnLoaded" against T if < 3 seconds
Which works. Sort of.

**Comment by Zyzyl** [comment]
What would the use case for this be? What would be the important distinction between OnWorldLoaded and OnLoaded/OnStart?

**Comment by 3x1t-5tyl3** [comment]
I'd like to differentiate between the world load and the item load:

My tool currently should only do certain things on item load and not on world load; but the "onloaded" event also triggers on worldload.

This would just significantly simplify certain things and I think there's plenty of other usecases where you'd wanna fire on world load or item load seperately.

**Comment by Frooxius** [comment]
Hmm... This might be perhaps better handled by getting the loading context from the event. It sounds like you want to do something when the item loads, but not when the world loads. With this impulse you wouldn't be able to do that directly, OnLoaded will fire in both cases still.

What might help there is getting the root of the loaded entity. This will give you more overall flexibility with things and you can check if the entity is a world or slot and potentially other entities.

**Comment by shadowpanther** [comment]
> What might help there is getting the root of the loaded entity. This will give you more overall flexibility with things and you can check if the entity is a world or slot and potentially other entities.

If I'm reading this right, this would allow to distinguish item load, avatar (with item in the backpack) load and item as a part of the world load and allow to customize behaviors for each case.

**Comment by 3x1t-5tyl3** [comment]
Sort. Just different events other than a generic "load" that would help in further checking for cases which allow for certain functions; example:

Run certain ref-setups on a tool; but only if it's spawned from the inventory; not on world load as the tool might've gotten certain overrides manually and I'd like to keep them

Setting of certain cloudvars but only if the world loads and not just the item; so it could be used to say "I loaded world x at y time" and the cloudvare'd item that is the same as in the world can be laoded instead but not overriden by being spawned out of the inventory.

Just to give 2 examples

**Comment by Frooxius** [comment]
Yes. With the proposed solution you get which is the root of the object that actually got loaded and that will let you decide - you could check if it's the avatar, the root of the item itself, the world or any other potential loadable entity.

**Comment by 3x1t-5tyl3** [comment]
Yeah that'd work perfectly.
